### DISCUSSION TITLE
Manual for dev environment setup

### DISCUSSION THREAD

**Discussion by Yinimi** [Caliburn-Micro/Caliburn.Micro#925] [Ideas]
I was struggling a bit to get the solution compiling since it supports some platforms I usually don't work with.
What do you think if we add a small section in the readme on how to set the development environment up and where to find what, so that you are ready to go.
This should make it easier for newcomers to contribute.
I'm thinking about something like this:

---

### Setup

To set everything up with Visual Studio:
* Install VS
* Enable the following features (workloads) in the vs installer: [...]
* Enable the following Individual components in the vs installer: [...]
* Install the following workloads: [...]
* Install Windows 10 SDK 10.0.19041 [sdk-archive](https://developer.microsoft.com/en-us/windows/downloads/sdk-archive/)

Setup with Rider:
* ?

Setup with vs code:
* ?

### Get started 

There are the following solutions
* main solution (path)
  Contains the code of the framework
* features solution (path)
  ...
* [...]

---

This might help to get an overview about the repository. I do not remember all the steps to get it running, might be required to test the setting up on a blank system.

**Comment by Yinimi** [comment]
Additionally it seems to be required to compile the mail solution once in Release | Any CPU so the dlls can be found for features solution etc.

**Reply by Yinimi** [reply]
It seems this can also just be selected under Individual Components

**Reply by vb2ae** [reply]
The workflows in Visual Studio 2022 you need should be .net Mutil-platform App UI Development, .net desktop development, and Windows Application development.   I have .net 8 installed.   Under indivual component I have Xamarin (out of support) checked.

**Reply by vb2ae** [reply]
> Additionally it seems to be required to compile the mail solution once in Release | Any CPU so the dlls can be found for features solution etc.

Yes we look at the release dlls in the samples.

**Comment by vb2ae** [comment]
What errors are you seeing when you try and build?

**Reply by vb2ae** [reply]
We should add a visual studio setup section to the documentation

**Reply by Yinimi** [reply]
I managed to fix everything, but sometimes it's not clear whats missing
